python/rmm/rmm/_lib/memory_resource.pyx

```diff
@@ -366,6 +378,49 @@ cdef class ManagedMemoryResource(DeviceMemoryResource):
         pass
 
 
+cdef class SystemMemoryResource(DeviceMemoryResource):
+    def __cinit__(self):
+        self.c_obj.reset(
+            new system_memory_resource()
+        )
+
+    def __init__(self):
+        """
+        Memory resource that uses ``malloc``/``free`` for
+        allocation/deallocation.
+        """
+        pass
+
+
+cdef class SamHeadroomResourceAdaptor(DeviceMemoryResource):
+    def __cinit__(
+        self,
+        size_t headroom
+    ):
+        self.system_mr = SystemMemoryResource()
+        self.c_obj.reset(
+            new sam_headroom_resource_adaptor[system_memory_resource](
```

Comment:
As above, not necessary to fix in this PR, but is there a reason to specialize the headroom adaptor for SAM vs supporting arbitrary upstreams? The Python resource is now specialized beyond the C++ resource, which is a potential source of confusion.

Reply:
Changed.